### ISSUE TITLE
Fix regression when extending rules in Less

### ISSUE THREAD

**Issue by glutentag** [stylelint/stylelint#3765]
When extending rules with Less, some errors are flagged.

> Which rule, if any, is the bug related to?

`property-no-unknown` and `declaration-colon-space-after`

> What CSS is needed to reproduce the bug?

```css
.a {
    &:extend(.b);
}
```

> What stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
  "rules": {
    "property-no-unknown": true,
    "declaration-colon-space-after": "always",
  }
}
```

> Which version of stylelint are you using?

`9.7.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

`stylelint test.less --syntax less`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's related to Less

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

The following warnings were flagged:

```shell
test.css
   2:5   ✖  Unexpected unknown property "&"   property-no-unknown          
   2:7   ✖  Expected single space after ":"   declaration-colon-space-after
```

**Comment by jeddy3** [comment]
@glutentag Thanks for the report.

This will need some digging to work out whether this is something we can fix in stylelint or whether it'll need to be addressed in the [Less parser itself](https://github.com/shellscape/postcss-less).

**Comment by jwilsson** [comment]
I [filed an issue](https://github.com/shellscape/postcss-less/issues/123) for it over at the parser repo, to clarify if the current behaviour is intentional or not.
